cmd/rm-main.go

```diff
@@ -242,6 +242,11 @@ func checkRmSyntax(ctx context.Context, cliCtx *cli.Context, encKeyDB map[string
 		fatalIf(errDummy().Trace(),
 			"This operation results in site-wide removal of objects. If you are really sure, retry this command with ‘--dangerous’ and ‘--force’ flags.")
 	}
+	if isNamespaceRemoval && (!isDangerous || !isForce) {
```

Comment:
```suggestion
	if isNamespaceRemoval && !(isDangerous && isForce) {
```
This is more readable to me (if is namespace removal and not is dangerous and force)

Reply:
This overlaps with the checks above.  It seems we are just throwing more cases into the mix.

Can't we simplify these?

Reply:
@klauspost @vadmeste  PTAL